### ISSUE TITLE
Global action

### ISSUE THREAD

**Issue by vanhtuan0409** [mirrorjs/mirror#59]
Is it possible to have an event that can be listened by multiple modules 😄 
For example, I have a listener that fire action every time screen media query changed. This action needs to be listened by multiple ui state modules

**Comment by llh911001** [comment]
Yes of course. There is a [`mirror.hook`](https://github.com/mirrorjs/mirror/blob/master/docs/api.md#mirrorhookaction-getstate--) API allows you to listen to every action you dispatched.

For your example, here is a possible solution:

```js

import mirror, { actions } from 'mirrorx'

mirror.model({
  name: 'mediaQuery',
  initialState: {
    width: 0,
    height: 0
  },
  reducers: {
    change(state, data) {
      return { ...state, ...data }
    }
  }
})

// in your media query change event
actions.mediaQuery.change({ width, height })


// in your models need to detect media query change
mirror.hook((action, getState) => {
  if (action.type === 'mediaQuery/change') {
    // do your work
  }
})
```

**Comment by vanhtuan0409** [comment]
This is what I doing right now
But can 1 model reducers listen to action of another model
Like can `stateOfComp` model listen to action from `mediaQuery` model

**Comment by llh911001** [comment]
You can. Try the [`mirror.hook`](https://github.com/mirrorjs/mirror/blob/master/docs/api.md#mirrorhookaction-getstate--) API, **though you should be careful to update state in hooks**.
